pkg/alert/commands.go

```diff
@@ -90,49 +93,86 @@ func NewGetAlertSettingsCommand() *cobra.Command {
 //NewAlertSettingGroupCommand creates the `alert setting` command
 func NewAlertSettingGroupCommand() *cobra.Command {
 	var (
-		id     int
-		enable bool
+		id         int
+		mustEnable bool
 	)
 
-	cmd := &cobra.Command{
+	root := &cobra.Command{
 		Use:              "setting",
-		Short:            "Print or enable a specific alert setting based on ID",
-		Example:          "alert setting --id=1001 [--enable]",
+		Short:            "Print an alert's settings",
+		Example:          "alert setting --id=1001",
 		TraverseChildren: true,
 		SilenceErrors:    true,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			if cmd.Flags().Changed("enable") {
-				if err := config.Client.EnableAlertSetting(id, enable); err != nil {
+			if mustEnable {
+				if err := config.Client.EnableAlertSetting(id, mustEnable); err != nil {
 					return err
 				}
 
-				if enable {
-					return bite.PrintInfo(cmd, "Alert setting [%d] enabled", id)
-				}
-				return bite.PrintInfo(cmd, "Alert setting [%d] disabled", id)
+				return bite.PrintInfo(cmd, "Alert setting [%d] enabled", id)
 			}
 
 			settings, err := config.Client.GetAlertSetting(id)
 			if err != nil {
-				golog.Errorf("Failed to retrieve alert [%d]. [%s]", id, err.Error())
 				return err
 			}
 
-			// force json, may contains conditions that are easier to be seen in json format.
 			return bite.PrintObject(cmd, settings)
 		},
 	}
 
-	cmd.Flags().IntVar(&id, "id", 0, "--id=1001")
-	cmd.MarkFlagRequired("id")
+	root.Flags().IntVar(&id, "id", 0, "--id=1001")
+	root.MarkFlagRequired("id")
 
-	cmd.Flags().BoolVar(&enable, "enable", false, "--enable")
+	root.Flags().BoolVar(&mustEnable, "enable", false, "--enable")
 
-	bite.CanPrintJSON(cmd)
-	bite.CanBeSilent(cmd)
+	bite.CanPrintJSON(root)
+	bite.CanBeSilent(root)
+
+	root.AddCommand(NewUpdateAlertSettingsCommand())
+	root.AddCommand(NewGetAlertSettingConditionsCommand())
+	root.AddCommand(NewAlertSettingConditionGroupCommand())
 
-	cmd.AddCommand(NewGetAlertSettingConditionsCommand())
-	cmd.AddCommand(NewAlertSettingConditionGroupCommand())
+	return root
+}
+
+// NewUpdateAlertSettingsCommand updates an alert's settings, e.g. channels, etc.
+func NewUpdateAlertSettingsCommand() *cobra.Command {
+	var alertSettings api.AlertSettingsPayload
+
+	cmd := &cobra.Command{
+		Use:              "set",
+		Short:            "Update an alert's settings or load from file. If `enable` parameter omitted, it defaults to true",
+		Example:          "alert setting set --alert=1001 --enable=true --channel='b83c862c-7e23-4fb4-863d-c03e04102f90' or alert setting set ./alert_sett.yml`",
+		TraverseChildren: true,
+		SilenceErrors:    true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Limitation: If `enable` not set through YAML,
```

Comment:
@deanlinaras should we remove this?

Reply:
It's a current limitation we have, good for other developers when revisiting this area. I'm afraid it might get lost if i remove this.